### PR TITLE
Improve search results ordering

### LINKED ISSUE #52: Improve search results ordering
The current ordering is only alphabetical.

There should be a sorting algorithm that priories matches according to the position of the match in the variable name (the earlier the better).

### PR DESCRIPTION
**Pull request by cbenz**
Fixes #52

### CODE REVIEW

File: src/components/pages/home.jsx [hunk 1]
```diff
@@ -112,15 +112,13 @@ const SearchResults = React.createClass({
...
         {parameterOrVariable => {
-          const {name, type} = parameterOrVariable
-          const description = type === 'parameter'
-            ? parameterOrVariable.description
-            : parameterOrVariable.label
+          const {description, label, name} = parameterOrVariable
+          const displayedDescription = description || label
           return (
-            <Link key={`${name}-${type}`} to={`/${name}`}>
+            <Link key={`${name}`} to={`/${name}`}>
```
Comment (MattiSG) [inline src/components/pages/home.jsx]: > Why interpolate? Wouldn't `key={name}` work? 🤔

File: src/search.js [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
   if (isEmpty(normalizedQuery)) {
     return sortBy(prop('name'), parametersAndVariables)
   }
-  return sortWith([
-    ascend(item => weightIn(item.name, normalizedQuery)),
-    descend(item => item.normalizedDescription.includes(normalizedQuery)),
-    ascend(prop('name')),
-  ], parametersAndVariables)
+  const matchesInName = pipe(
+    filter(item => item.name.includes(normalizedQuery)),
```
Comment (MattiSG) [inline src/search.js]: Use [`splitWhen`](http://ramdajs.com/docs/#splitWhen) to avoid iterating twice on the input set :wink:

**Comment by MattiSG** [comment]
There's an issue with the commit message for 6c74fa4.

File: src/components/pages/home.jsx (hunk 1 above)
Comment (cbenz) [inline src/components/pages/home.jsx]: Haha, this is dumb indeed. But after more thought I think this change was a mistake. It's *possible* as we said, that a variable and a parameter have the same name. That's why I added a `.type == 'variable' (or 'parameter')` attribute.

I must fix that.

File: src/search.js (hunk 2 above)
Comment (cbenz) [inline src/search.js]: That was [`partition`](http://ramdajs.com/docs/#partition) actually (`splitWhen` stops partitioning at first match). But thanks for the good idea!

**Comment by MattiSG** [comment]
Please don't force-push while in the review process. This makes it impossible to track changes since last review, thus making the reviewer's job very hard.

**Pull request merged**